examples/dmd/de_parametric_heat_conduction_greedy.cpp

```diff
@@ -748,28 +748,28 @@ double simulation()
 
         delete result_u;
 
-        for (int i = 1; i < ts.size(); i++)
+        if (visit)
         {
-            result_u = dmd_u->predict(ts[i]);
-            if (myid == 0)
+            for (int i = 1; i < ts.size(); i++)
             {
-                std::cout << "Predicting temperature using DMD at: " << ts[i] << std::endl;
-            }
+                result_u = dmd_u->predict(ts[i]);
+                if (myid == 0)
+                {
+                    std::cout << "Predicting temperature using DMD at: " << ts[i] << std::endl;
+                }
 
-            Vector dmd_solution_u(result_u->getData(), result_u->dim());
-            u_gf.SetFromTrueDofs(dmd_solution_u);
+                Vector dmd_solution_u(result_u->getData(), result_u->dim());
+                u_gf.SetFromTrueDofs(dmd_solution_u);
 
-            if (i == ts.size() - 1 || (i % vis_steps) == 0)
-            {
-                if (visit)
+                if (i == ts.size() - 1 || (i % vis_steps) == 0)
```

Comment:
This condition could include everything in the `i`-loop. There is no need to compute the DMD prediction if `i` is not a VisIt output step. The same goes for the other examples.

Reply:
Instead of including the condition in for loop, I have moved the if statement right under the main time for loop for dmd evaluation. I was not sure how to include "i == ts.size() - 1" in the for loop.